eZ/Bundle/EzPublishCoreBundle/Imagine/VariationPathGenerator/WebpFormatVariationPathGenerator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator;
+
+use eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator;
+use Liip\ImagineBundle\Imagine\Filter\FilterConfiguration;
+
+/**
+ * Decorates VariationPathGenerator with .webp extension if image variation is configured for this format.
+ */
+final class WebpFormatVariationPathGenerator implements VariationPathGenerator
+{
+    /** @var \eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator */
+    private $innerVariationPathGenerator;
+
+    /** @var \Liip\ImagineBundle\Imagine\Filter\FilterConfiguration */
+    private $filterConfiguration;
+
+    public function __construct(
+        VariationPathGenerator $innerVariationPathGenerator,
+        FilterConfiguration $filterConfiguration
+    ) {
+        $this->innerVariationPathGenerator = $innerVariationPathGenerator;
+        $this->filterConfiguration = $filterConfiguration;
+    }
+
+    public function getVariationPath($originalPath, $filter)
+    {
+        $variationPath = $this->innerVariationPathGenerator->getVariationPath($originalPath, $filter);
+        $filterConfig = $this->filterConfiguration->get($filter);
+
+        if (!isset($filterConfig['format']) || $filterConfig['format'] !== 'webp') {
+            return $variationPath;
+        }
+
+        return $variationPath . '.webp';
```

Comment:
won't this result with e.g. `image.jpg.webp` instead of `image.webp`

Reply:
This is correct. It looks weird but `webp` format actually uses double extension: https://symfony.com/bundles/LiipImagineBundle/current/basic-usage.html#use-webp-if-supported